lib/index.js

```diff
@@ -1,11 +1,19 @@
 'use strict';
 
+const RateLimiter = require('./rate-limiter');
+
 exports.register = (server, options, next) => {
 
   server.ext('onPostAuth', (request, reply) => {
-    request.payload.loaded = true;
+    return RateLimiter.limit(request, options.redisClient, options.defaultRate)
+    .then((rate) => {
+      if (rate && rate.current > rate.limit) {
+        return reply(options.overLimitHandler());
+      }
 
-    return reply.continue();
+      request.payload.rate = rate;
```

Comment:
According to the Hapi docs you should save any plugin-specific information in `request.plugins['hapi-rate-limit']`

> plugins - plugin-specific state. Provides a place to store and pass request-level plugin data. The plugins is an object where each key is a plugin name and the value is the state.

http://hapijs.com/api#request-object